packages/liveblocks-core/src/lib/guards.ts

```diff
@@ -0,0 +1,18 @@
+export function isPlainObject(
+  blob: unknown
+): blob is { [key: string]: unknown } {
+  // Implementation borrowed from pojo decoder, see
+  // https://github.com/nvie/decoders/blob/78849f843193647eb6b5307240387bdcff7161fb/src/lib/objects.js#L10-L41
+  return (
+    blob !== null &&
+    typeof blob === "object" &&
+    Object.prototype.toString.call(blob) === "[object Object]"
+  );
+}
+
+/**
+ * Check if value is of shape { startsWith: string }
+ */
+export function isStartsWith(blob: unknown): blob is { startsWith: string } {
```

Comment:
Tiny nitpick but maybe we could name it `isStartsWithOperator` or similarly? Or do you want to keep it not specific to operators? At a glance it's pretty similar to `startsWith()`.

Reply:
Agreed, that's actually a better name. Will rename before merging! 👍